trieste/acquisition/multi_objective/partition.py

```diff
@@ -45,35 +49,48 @@ def prepare_default_non_dominated_partition_bounds(
         shape.
+    :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``anti_reference`` has an invalid
+        shape.
     """
-    assert tf.shape(observations)[-1] >= 2, ValueError(
-        "the last dimension of multi-objective "
-        f"observations (i.e., objective number) is expected to be at "
-        f"least 2 but found {tf.shape(observations)[-1]}"
-    )
-    if (
-        anti_reference is None
-    ):  # if anti_reference point is not specified, use a -1e10 as default (act as -inf)
-        # make sure given observations are larger than -1e10
-        anti_reference = -1e10 * tf.ones(
-            shape=(tf.shape(observations)[-1]), dtype=observations.dtype
-        )
-        tf.debugging.assert_greater_equal(
-            observations,
-            anti_reference,
-            message="observations containing points below default "
-            "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
-            "anti-reference point.",
-        )
+
+    def not_valid_obs(obs: Optional[TensorType]) -> bool:
+        return obs is None or tf.equal(tf.size(observations), 0)
+
+    def specify_default_anti_reference_point(
+        ref: TensorType, obs: Optional[TensorType]
+    ) -> TensorType:
+        anti_ref = -1e10 * tf.ones(shape=(tf.shape(reference)), dtype=reference.dtype)
         tf.debugging.assert_greater_equal(
-            reference,
-            anti_reference,
+            ref,
+            anti_ref,
             message="reference point containing dimensionality below default "
             "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
             "anti-reference point.",
         )
+        if not not_valid_obs(obs):  # make sure given (valid) observations are larger than -1e10
+            tf.debugging.assert_greater_equal(
+                obs,
+                anti_ref,
+                message="observations containing points below default "
+                "anti-reference point ([-1e10, ..., -1e10]), try specify a lower "
+                "anti-reference point.",
+            )
+        return anti_ref
 
-    if tf.shape(observations)[-1] > 2:
+    tf.debugging.assert_shapes([(reference, ["D"])])
+    if (  # prepare default anti_reference point if not specified
+        anti_reference is None
+    ):  # if anti_reference point is not specified, use a -1e10 as default (act as -inf)
+        anti_reference = specify_default_anti_reference_point(reference, observations)
+    else:  # anti_reference point is specified
+        tf.debugging.assert_shapes([(anti_reference, ["D"])])
+
+    if not_valid_obs(observations):  # if no valid observations
+        assert tf.reduce_all(tf.less_equal(anti_reference, reference)), ValueError(
+            "anti_reference points contains dimension bigger than reference point"
+        )
+        return tf.expand_dims(anti_reference, 0), tf.expand_dims(reference, 0)
```

Comment:
is it OK that this doesn't return a `_BoundIndexPartition`?

Reply:
Was originally expecting calling `prepare_default_non_dominated_partition_bounds` is the same as calling a `any_partition().partition_bounds()`. The latter is expected to return just a tuple of _static_ `lb_points` and `ub_points`.    
So as long as a _static_ `lb_points` & `ub_points` is acceptable for acq, not return a `_BoundIndexPartition_instance`/`any_partition_instance` is ok?

Reply:
Yes, we only provide bounds here. Not immediately aware of a case we need `partition_instance`, unless we need somehow dynamically change partition bounds on the fly during one iteration (if having this use case please indicate), otherwise maybe we can just provide _static_ `partition_bounds` as a "default" choice.